pkg/e2e-test.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+#
+# Copyright 2022 The Sigstore Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+testdir=$(dirname "$0")
+
+rm -f /tmp/pkg-rekor-*.cov
+echo "installing gocovmerge"
+make gocovmerge
+docker kill $(docker ps -q) || true
+echo "starting services"
+docker-compose -f docker-compose.yml -f docker-compose.test.yml up -d --build
+
+echo "building CLI and server"
+# set the path to the root of the repo
+dir=$(git rev-parse --show-toplevel)
+go test -c ./cmd/rekor-cli -o rekor-cli -cover -covermode=count -coverpkg=./...
+go test -c ./cmd/rekor-server -o rekor-server -covermode=count -coverpkg=./...
+
+count=0
+echo -n "waiting up to 120 sec for system to start"
+until [ $(docker-compose ps | grep -c "(healthy)") == 3 ];
+do
+    if [ $count -eq 12 ]; then
+       echo "! timeout reached"
+       exit 1
+    else
+       echo -n "."
+       sleep 10
+       let 'count+=1'
+    fi
+done
+
+echo
+echo "running tests"
+REKORTMPDIR="$(mktemp -d -t rekor_test.XXXXXX)"
+cp $dir/rekor-cli $REKORTMPDIR/rekor-cli
+touch $REKORTMPDIR.rekor.yaml
+trap "rm -rf $REKORTMPDIR" EXIT
+if ! REKORTMPDIR=$REKORTMPDIR go test -tags=e2e ./pkg/...; then
+   docker-compose logs --no-color > /tmp/docker-compose.log
+   exit 1
+fi
+if docker-compose logs --no-color | grep -q "panic: runtime error:" ; then
+   # if we're here, we found a panic
+   echo "Failing due to panics detected in logs"
+   docker-compose logs --no-color > /tmp/docker-compose.log
+   exit 1
+fi
+
+echo "generating code coverage"
+curl -X GET 0.0.0.0:2345/kill
+sleep 5
```

Comment:
Something we need to consider, is that part of the e2e tests have dependencies on specific entries; the Rekor instance is launched without specifying a particular TreeID (for these tests), and therefore as you break this apart you will need to think through which entries a test needs in order to be successful.

Reply:
Since we kill the instance to gather code coverage here, the entries are lost unless in the future we change these tests to start Rekor targeted at a specific treeID.

Reply:
So should we populate the rekor with specific entries before the tests? How is it working for the existing e2e? Is this a missing feature at present?

Reply:
At the moment, this is more of an FYI rather than something that's directly actionable. Just something for you to be aware of as you continue moving things over, is that unless we change the test harness to be more shard aware, you may run into issues around tests that are dependent on one another. We should probably make those explicit (with wrapping and use of `t.Run`) whenever we see them.

Reply:
OK, Thanks. What else is required to get this merged in?

Reply:
tests need to pass (I just re-triggered them).